compute/cloud-client/src/main/java/compute/CreateInstance.java

```diff
@@ -88,21 +89,21 @@ public static void createInstance(String project, String zone, String instanceNa
       // Insert the instance in the specified project and zone.
       Operation response = instancesClient.insert(project, zone, instanceResource);
 
-      if (response.getStatus() == Status.RUNNING) {
-        // Wait for the create operation to complete; default timeout is 2 mins
-        response = zoneOperationsClient.wait(project, zone, String.valueOf(response.getId()));
+      // Wait for the create operation to complete.
+      // Timeout is set at 180000ms or 3 minutes.
+      long startTime = System.currentTimeMillis();
+      while (response.getStatus() != Status.DONE
+          && System.currentTimeMillis() - startTime < 180000) {
```

Comment:
[Please use the `java.time` package here.](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#time)
```suggestion
      LocalTime endTime = LocalTime.now().plusMinutes(3);
      while (response.getStatus() != Status.DONE && LocalTime.now().isBefore(endTime) {
```

Reply:
same elsewhere

Reply:
Changed it. Thanks!